code/modules/mob/living/simple_animal/abnormality/waw/dreaming_current.dm

```diff
@@ -103,10 +114,20 @@
 		var/turf/T = get_closest_atom(/turf/open, potential_turfs, picking_from)
 		if(!T)
 			break
-		movement_path += get_path_to(path_start, T, /turf/proc/Distance_cardinal)
+		var/list/our_path = list()
+		for(var/o = 1 to 3) // Grand total of 3 retries
+			our_path = get_path_to(path_start, T, /turf/proc/Distance_cardinal, dash_max_distance)
+			if(islist(our_path) && LAZYLEN(our_path))
+				break
+			potential_turfs -= T // Couldn't find path to it, don't try again
```

Comment:
Wouldn't this result in a duplicate removal if it initially could not get a path, but later managed to on, say loop 2 or 3?

Reply:
Right, I will have to pick a new T variable here.